### PR TITLE
Added Circle in the Dgame.Math

### PR DESCRIPTION
**Pull request by leotada**
Added Circle math, with colliding detection with other circle. Some things are missing comparing with Rect.

### CODE REVIEW

File: source/Dgame/Math/Circle.d [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+private:
+
+import derelict.sdl2.sdl;
+
+import Dgame.Math.Vector2;
+
+import std.math: sqrt;
+
+package(Dgame):
```
Comment (Dgame) [inline source/Dgame/Math/Circle.d]: I think this is superfluous.

File: source/Dgame/Math/Circle.d [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+     * If, and the parameter 'overlap' isn't null,
+     * the colliding circle is stored there.
+     */
+    @nogc
+    bool intersects(const Circle circle, Circle* overlap = null) const {
+        Vector2i center1 = getCenter();
+        Vector2i center2 = circle.getCenter();
+        int dx = center1.x - center2.x;
+        int dy = center1.y - center2.y;
+        float distance = sqrt(cast(float)dx * dx + dy * dy);
```
Comment (Dgame) [inline source/Dgame/Math/Circle.d]: This whole bunch of variables could be immutable.

File: source/Dgame/Math/Circle.d [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+        if (distance < this.radius + circle.radius) {
+            return true;
+        }
+        else {
```
Comment (Dgame) [inline source/Dgame/Math/Circle.d]: That's not the general format.

Comment (Dgame) [inline source/Dgame/Math/Circle.d]: The `else` is superfluous.

File: source/Dgame/Math/Circle.d (hunk 1 above)
Comment (leotada) [inline source/Dgame/Math/Circle.d]: sqrt is used in collision detection. I didn't understand.

File: source/Dgame/Math/Circle.d (hunk 3 above)
Comment (leotada) [inline source/Dgame/Math/Circle.d]: How do this?

File: source/Dgame/Math/Circle.d (hunk 1 above)
Comment (Dgame) [inline source/Dgame/Math/Circle.d]: I mean line 34 `package(Dgame):`

File: source/Dgame/Math/Circle.d (hunk 3 above)
Comment (Dgame) [inline source/Dgame/Math/Circle.d]:
```d
if (distance < this.radius circle.radius) {
    return true;
}

return false;
```

or even simpler

```d
return distance < this.radius circle.radius
```

**Comment by Dgame** [comment]
LGTM

**Pull request merged**